base/server/src/main/java/com/netscape/cms/realm/PKIPostgreSQLRealm.java

```diff
@@ -119,17 +119,27 @@ public void initInternal () throws LifecycleException {
         if (saltLength != null) {
             handler.setSaltLength(Integer.parseInt(saltLength));
         }
+
+        String createFile = info.getProperty("dbcreate.file");
+        if (createFile != null) {
+            try{
+                connect();
+                setup(createFile);
+            } catch (Exception e) {
+                throw new LifecycleException("DB creation failed. Creation file: " + createFile, e);
+            }
+        }
+        
```

Comment:
I think the `setup()` was originally called from `connect()` because there is no guarantee that containers will be started in certain orders. For example, sometimes the ACME container is started first, then the PostgreSQL container is added later or it was not running when ACME container is started. With the new code that probably won't work anymore. My suggestion is to return `setup()` to its original position, but we can keep the `dbcreate.file` param. What do you think?

Reply:
I do not think this parameter should be used after the installation. If we have it in the init then at the server start the DB should be created and the parameter should be removed. This could be done during pkispawn by the user.

In a container environment there is no reason to provide a DB without table after the initial setup. Who should fill the table with the user? In any case it should be possible to define a startup order although it is not needed in this case.

Since the current implementation perform the check in the `connect()` I am reverting it and evaluate again when we will move to container.

Reply:
Thanks! Yeah, ACME doesn't require any user unless we want to manage something remotely (which there isn't much that can be done anyway). EST seems to require a pre-existing realm, so the requirement is different. We can certainly revisit this requirement again.